pool/hashdata_test.go

```diff
@@ -0,0 +1,132 @@
+package pool
+
+import (
+	"errors"
+	"math/big"
+	"testing"
+	"time"
+
+	errs "github.com/decred/dcrpool/errors"
+)
+
+func testHashData(t *testing.T) {
+	miner := ObeliskDCR1
+	extraNonce1 := "ca750c60"
+	ip := "127.0.0.1:5550"
+	now := time.Now()
+	hashRate := new(big.Rat).SetInt64(100)
+
+	hashData := newHashData(miner, xID, ip, extraNonce1, hashRate)
+
+	// Ensure hash data can be persisted.
+	err := db.persistHashData(hashData)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure hash data can be fetched.
+	hashID := hashDataID(xID, extraNonce1)
```

Comment:
imo it makes more sense to retrieve using `hashData.UUID`, rather than recreating the ID manually using the `hashDataID` func. Thats how the other tests are working.